### PR TITLE
 Sync v2-7-stable with v2-7-test to release 2.7.0b1 

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.7.0b1`

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,258 @@
...
+- Add multi-select, proposals and labels for trigger form (#31441)
+- Making webserver config customizable (#29926)
+- Render DAGCode in the Grid View as a tab (#31113)
+- Add rest endpoint to get option of configuration (#31056)
+- Add ``section`` query param in get config rest API (#30936)
+- Create metrics to track ``Scheduled->Queued->Running`` task state transition times (#30612)
+- Mark Task Groups as Success/Failure (#30478)
+- Add CLI command to list the provider trigger info (#30822)
+- Add Fail Fast feature for DAGs (#29406)
+- Adding configuration to control retry parameters for k8s api client (#29809)
```
Comment (eladkal) [inline RELEASE_NOTES.rst]: This is listed under 2.6.0 is it a wrong entry here or did it miss 2.6.0?

**Review by eladkal — COMMENTED** [review]
General comment about the release notes..
It contains many K8s and Celery changes that when merged were part of core but now they have already been released with last Providers wave. I think having them listed in core release will be very confusing to users.

WDYT?

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,258 @@
...
+- Support ``Pydantic`` 2 (#32366)
+- Limit click until we fix mypy issues (#32413)
+- A couple of minor cleanups (#31890)
+- Replace State usages with strong-typed ``enums`` (#31735)
+- Upgrade ruff to ``0.272`` (#31966)
+- Better error message when serializing callable without name (#31778)
+- Improve the views module a bit (#31661)
+- Remove ``asynctest`` (#31664)
+- Refactor sqlalchemy queries to ``2.0`` style (#31569, #31772, #32350, #32339, #32474, #32645)
+- Remove Python ``3.7`` support (#30963)
```
Comment (eladkal) [inline RELEASE_NOTES.rst:298]: I think dropping Python 3.7 should be under significant changes.
It's very important note to many users.

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst]: Yeah. Getting changelog for minor releases is always a pain. I will remove that

**Comment by ephraimbuddy** [comment]
> General comment about the release notes.. It contains many K8s and Celery changes that when merged were part of core but now they have already been released with last Providers wave. I think having them listed in core release will be very confusing to users.
> 
> WDYT?

It's important to also have it at the core changelog because it contains a lot of important code changes. Users that are not able to import certain packages where they'd previously imported them can easily find the change in the core

**Comment by potiuk** [comment]
> > General comment about the release notes.. It contains many K8s and Celery changes that when merged were part of core but now they have already been released with last Providers wave. I think having them listed in core release will be very confusing to users.
> > WDYT?
> 
> It's important to also have it at the core changelog because it contains a lot of important code changes. Users that are not able to import certain packages where they'd previously imported them can easily find the change in the core

Agree, we should be quite vocal in release notes about it. This will be a bit surprise to many people but this is hopefully the only time we have to do it and there has to be one release when we "transition" - and almost by definition this means that the changes have to be mentioned on both sides.

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,258 @@
...
 
+Airflow 2.7.0 (2023-08-10)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+Disable default allowing the testing of connections in UI, API and CLI(#32052)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+The test connection functionality is disabled by default across Airflow UI,
```
Comment (potiuk) [inline RELEASE_NOTES.rst]:
```suggestion
For security reasons, the test connection functionality is disabled by default across Airflow UI,
```

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,6 +21,258 @@
...
+The following values are accepted for this config param:
+1. ``Disabled``: Disables the test connection functionality and
+disables the Test Connection button in the UI.
+This is also the default value set in the Airflow configuration.
+2. ``Enabled``: Enables the test connection functionality and
+activates the Test Connection button in the UI.
+3. ``Hidden``: Disables the test connection functionality and
+hides the Test Connection button in UI.
+For more information on capabilities of users, see the documentation:
+https://airflow.apache.org/docs/apache-airflow/stable/security/index.html#capabilities-of-authenticated-ui-users
```
Comment (potiuk) [inline RELEASE_NOTES.rst]: This has now moved to a different URL. I think it should be:


https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html#capabilities-of-authenticated-ui-users

Comment (potiuk) [reply RELEASE_NOTES.rst]: Tested it with locally built docs.

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,6 +21,258 @@
...
+
```
Comment (potiuk) [inline RELEASE_NOTES.rst:73]:
```suggestion
It is strongly advised to **not** enable the feature until you make sure that only 
highly trusted UI/API users have "edit connection" permissions.
```

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,6 +21,258 @@
...
+in which case it'll always use the value of ``core.parallelism``.
+
+The Kubernetes, Celery, CeleryKubernetes, LocalKubernetes, and Dask executors are moved to corresponding providers (#32767)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+In order to use the executors, you need to install the providers:
+
+* for Celery executors you need to install ``apache-airflow-providers-celery`` package >= 3.3.0
+* for Kubernetes executors you need to install ``apache-airflow-providers-cncf-kubernetes`` package >= 7.4.0
+* For Dask executors you need to install ``apache-airflow-providers-daskexecutor`` package in any version
+
```
Comment (potiuk) [inline RELEASE_NOTES.rst:114]:
```suggestion
You can achieve it also by installing airflow with ``[celery]``, ``[cncf.kubernetes]``, ``[daskexecutor]`` extras respectively.

Users who base their images on the ``apache/airflow`` reference image (not slim) should be unaffected - the base
reference image comes with all the three providers installed.
```

File: RELEASE_NOTES.rst (hunk 2 above)
Comment (potiuk) [reply RELEASE_NOTES.rst:298]: Yeah. I think we should also mention that the reference ``apache/airflow`` image is 3.8.

Comment (potiuk) [reply RELEASE_NOTES.rst:298]: This should be done in `docs/docker-stack/changelog.rst`

Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:298]: Done

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
Small comment - we need #33070 in

**Review by potiuk — APPROVED** [review]

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,6 +21,296 @@
...
+This is also the default value set in the Airflow configuration.
+2. ``Enabled``: Enables the test connection functionality and
+activates the Test Connection button in the UI.
+3. ``Hidden``: Disables the test connection functionality and
+hides the Test Connection button in UI.
+For more information on capabilities of users, see the documentation:
+https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html#capabilities-of-authenticated-ui-users
+It is strongly advised to **not** enable the feature until you make sure that only
+highly trusted UI/API users have "edit connection" permissions.
+The ``xcomEntries`` API disables support for the ``deserialize`` flag by default (#32176)
```
Comment (potiuk) [inline RELEASE_NOTES.rst:74]: Empty line here?

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,6 +21,296 @@
...
+""""""""""""""""""""""""""""""""""
+As of now, Python 3.7 is no longer supported by the Python community.
+Therefore, to use Airflow 2.7.0, you must ensure your Python version is
+either 3.8, 3.9, 3.10, or 3.11.
+
+Old Graph View is removed (#32958)
+""""""""""""""""""""""""""""""""""
+The old Graph View is removed. The new Graph View is the default view now.
+
+In case of SMTP SSL connection, the default context now uses "default" context (#33070)
```
Comment (potiuk) [inline RELEASE_NOTES.rst]: In case of SMTP SSL connection, the context now uses "default" context (#33070)

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -21,6 +21,296 @@
...
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+The "default" context is Python's ``default_ssl_contest`` instead of previously used "none". The
```
Comment (potiuk) [inline RELEASE_NOTES.rst:42]: I think those paragraphs here would benefit from a little more spacing in general. This looks too much like a wall of text, splitting it into paragraphs and separating them with empty lines would be much more readable.

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -21,6 +21,296 @@
...
+API endpoint now disables the ``deserialize`` option to deserialize arbitrary
+XCom values in the webserver. For backward compatibility, server admins may set
+the ``[api] enable_xcom_deserialize_support`` config to *True* to enable the
+flag and restore backward compatibility.
+
+However, it is strongly advised to **not** enable the feature, and perform
+deserialization at the client side instead.
+
+Default name of the Celery application changed from ``airflow.executors.celery_executor`` to ``airflow.providers.celery.executors.celery_executor`` (#32526)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
Comment (potiuk) [inline RELEASE_NOTES.rst]: This likely should be first paragraph - but the header should be simpler: 

``Change of the celery app name``  ?

File: RELEASE_NOTES.rst [hunk 11]
```diff
@@ -21,6 +21,296 @@
...
+This change is expected to make the Scheduler more responsive.
+
+``scheduler.max_tis_per_query`` needs to be lower than ``core.parallelism``.
+If both were left to their default value previously, the effective default value of ``scheduler.max_tis_per_query`` was 32
+(because it was capped at ``core.parallelism``).
+
+To keep the behavior as close as possible to the old config, one can set ``scheduler.max_tis_per_query = 0``,
+in which case it'll always use the value of ``core.parallelism``.
+
+The Kubernetes, Celery, CeleryKubernetes, LocalKubernetes, and Dask executors are moved to corresponding providers (#32767)
```
Comment (potiuk) [inline RELEASE_NOTES.rst]: Same here. I think the header should be:

``Some executors have been moved to providers`` or smth like that.

**Review by potiuk — COMMENTED** [review]
Few NIT's when it comes to readability of the release notes.

**Pull request merged**